packages/api/cms-api/src/blocks/block-index.service.ts

```diff
@@ -1,51 +1,92 @@
+                            targetTableData->>'tableName'         "targetTableName",
+                            targetTableData->>'primary'           "targetPrimaryKey",
+                            dependenciesObj->>'id' "targetId"
                         FROM "${metadata.tableName}",
-                            json_array_elements("${metadata.tableName}"."${column}"->'index') index,
-                            json_array_elements_text("index"->'damFileIds') files`;
-            damFilesIndexSelects.push(select);
+                            json_array_elements("${metadata.tableName}"."${column}"->'index') indexObj,
+                            json_array_elements(indexObj->'target'->'dependencies') dependenciesObj,
+                            json_extract_path('${JSON.stringify(targetEntitiesNameData)}', dependenciesObj->>'targetEntityName') targetTableData`;
+
+            indexSelects.push(select);
         }
 
-        const viewSql = damFilesIndexSelects.join("\n UNION ALL \n");
+        const viewSql = indexSelects.join("\n UNION ALL \n");
 
-        console.time("creating block damFiles materialized view");
-        await this.connection.execute(`DROP MATERIALIZED VIEW IF EXISTS block_index_damfiles`);
-        await this.connection.execute(`CREATE MATERIALIZED VIEW block_index_damfiles AS ${viewSql}`);
-        console.timeEnd("creating block damFiles materialized view");
+        console.time("creating block dependency materialized view");
+        await this.connection.execute(`DROP MATERIALIZED VIEW IF EXISTS block_index_dependencies`);
+        await this.connection.execute(`CREATE MATERIALIZED VIEW block_index_dependencies AS ${viewSql}`);
+        console.timeEnd("creating block dependency materialized view");
 
-        console.time("creating block damFiles materialized view index");
-        await this.connection.execute(`CREATE INDEX block_index_damfiles_damfileid ON block_index_damfiles ("damFileId")`);
-        console.timeEnd("creating block damFiles materialized view index");
+        console.time("creating block dependency materialized view index");
+        await this.connection.execute(`CREATE INDEX block_index_dependencies_targetId ON block_index_dependencies ("targetId")`);
+        console.timeEnd("creating block dependency materialized view index");
     }
 
     async refreshViews(): Promise<void> {
-        console.time("refresh materialized block damFiles");
-        await this.connection.execute("REFRESH MATERIALIZED VIEW block_index_damfiles");
-        console.timeEnd("refresh materialized block damFiles");
+        console.time("refresh materialized block dependency");
+        await this.connection.execute("REFRESH MATERIALIZED VIEW block_index_dependencies");
+        console.timeEnd("refresh materialized block dependency");
+    }
+
+    async getDependentsByTargetEntityNameAndTargetId(targetEntityName: string, targetId: string): Promise<BlockIndexDependency[]> {
```

Comment:
would be `getDependentsByTarget` as name not enough?

Reply:
Theoretically, it would also be possible to use `tableName + id` or `graphqlObjectType + id` for identification. The availability of these properties may vary. Therefore, I wanted to retain the possibility to add getter methods for these combinations (`getDependentsByTargetTableNameAndTargetId` and getDependentsByTargetGraphqlObjectTypeAndTargetId`). But I didn't add them yet, since I'm not quite sure yet if we need them.

Reply:
In my opinion, three good naming options would be:

`getDependents()` and `getDependencies()`

- remove `ByTarget` and `ByRoot` because they are implicit (a target has dependents and a root has dependencies. It can be hard to wrap one's hand around that, thus I left target and root in the name to make it more obvious)
- implies getting by entityName is the default, but we could add other getter methods in the future and name them, for example, `getDependentsByTableName()`

`getDependentsByEntityName()` and `getDependenciesByEntityName()`

- target and root are implicit
- more methods can be added and it's still precise

`getDependentsByTargetEntityName()` and `getDependenciesByRootEntityName()`

- target and root are explicit -> easier to understand and use
- more methods can be added and it's still precise

Reply:
I prefer option 3 `getDependentsByTargetEntityName()` and `getDependenciesByRootEntityName()`, but I'm also okay with the other options. What do you think?

Reply:
Another option would be to put the target/root in the argument: `getDependents(target: { entityName: string; id: string })` and `getDependencies(root: { entityName: string; id: string })`. Would make target and root explicit by still keeping simple method names

Reply:
Yes, that's also a good option. I implemented it

Reply:
:+1: for @johnnyomair option

Reply:
And please consider finding a way to pass the entry object itself

Reply:
@nsams 
I already implemented that, but didn't use it (https://github.com/vivid-planet/comet/pull/842/files#diff-756b8a8301df7ea1c04ee25f9ed4726b2bb786bdb364746d4eb00fe824894f1dR80-R81). I used the same pattern as Mikroorm (https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts#L106-L110)

I now added a commit that actually uses this functionality.